### PR TITLE
moved to using appdirs for sunpyrc config

### LINKED ISSUE #2869: Use appdirs for our config file locations
https://pypi.org/project/appdirs/

### PR DESCRIPTION
**Pull request by nabobalis**
### Description

Fixes #2869

TODO: 

- [x] Write tests

### CODE REVIEW

File: sunpy/util/config.py [hunk 1]
```diff
@@ -24,40 +36,59 @@ def load_config():
...
 
 
+def _find_config_files():
+    """Finds locations of SunPy configuration files"""
+    config_files = []
+    config_filename = 'sunpyrc'
+
+    # find default configuration file
+    module_dir = Path(sunpy.__file__).parent
+    config_files.append(str(module_dir / 'data' / 'sunpyrc'))
```
Comment (dpshelio) [inline sunpy/util/config.py]: I believe this works:
```suggestion
    config_files.append(f"{module_dir / 'data' / 'sunpyrc'}")
```

File: sunpy/util/config.py [hunk 2]
```diff
@@ -24,40 +36,59 @@ def load_config():
...
-    filepaths = [
-        ('downloads', 'download_dir'),
-        ('downloads', 'sample_dir')
-    ]
-    _fix_filepaths(config, filepaths)
+    working_dir = Path(config.get('general', 'working_dir'))
+    sample_dir = Path(config.get('downloads', 'sample_dir'))
+    download_dir = Path(config.get('downloads', 'download_dir'))
+    config.set('downloads', 'sample_dir', str((working_dir / sample_dir).expanduser().resolve()))
+    config.set('downloads', 'download_dir', str((working_dir / download_dir).expanduser().resolve()))
```
Comment (dpshelio) [inline sunpy/util/config.py]: Can't config use `Path` directly?

File: sunpy/util/config.py (hunk 1 above)
Comment (dpshelio) [reply sunpy/util/config.py]: Similarly in other places.

File: sunpy/util/config.py [hunk 3]
```diff
@@ -78,54 +109,13 @@ def print_config():
...
-
-    return config_files
+    return Path(p).is_dir() and os.access(p, os.W_OK)
 
 
 def _get_user_configdir():
     """
     Return the string representing the configuration dir.
-    The default is "HOME/.sunpy".  You can override this with the
+    The default is "HOME/.sunpy". You can override this with the
```
Comment (dpshelio) [inline sunpy/util/config.py]: I think it makes more sense if it is `$HOME`
```suggestion
    The default is "$HOME/.sunpy". You can override this with the
```

File: sunpy/util/config.py [hunk 4]
```diff
@@ -1,11 +1,23 @@
...
+
+# This is to avoid creating a new config dir for each new dev version.
+# We use AppDirs to locate and create the config directory.
+dirs = AppDirs("sunpy", "sunpy")
+# Default one set by AppDirs
+CONFIG_DIR = dirs.user_config_dir
+
+# This is to avoid creating a new config dir for each new dev version.
+# We use AppDirs to locate and create the config directory.
+dirs = AppDirs("sunpy", "sunpy")
```
Comment (dpshelio) [inline sunpy/util/config.py]: Why is this repeated also on l14?

Comment (nabobalis) [reply sunpy/util/config.py]: Because I can't do a rebase merge correctly.

File: sunpy/util/config.py (hunk 2 above)
Comment (nabobalis) [reply sunpy/util/config.py]: That is a good question and one I did not plan to test out until we plan to move to `Path` in more files.

File: sunpy/util/config.py (hunk 1 above)
Comment (nabobalis) [reply sunpy/util/config.py]: I mean it works fine, it feels a little overkill to str something?

Comment (nitish705702) [reply sunpy/util/config.py]: i think, it work good,very little change f over str.

Comment (Cadair) [reply sunpy/util/config.py]: There is also `Path.as_posix()` for a third option.

I think `str()` is probably lower overhead on the interpreter and a little more familiar to people, but I don't really think it makes any real difference.

**Comment by nabobalis** [comment]
Seems that `hec.HECClient()` found no servers.

**Pull request merged**